File: app-tracker-automation/pages/dashboard_page.py
"""
Dashboard Page Object Model
Handles dashboard functionality for the insurance portal
"""

from playwright.sync_api import Page, expect
from pages.base_page import BasePage
from locators.shared_locators import SharedLocators


class DashboardPage(BasePage):
    """Dashboard page object"""
    
    def __init__(self, page: Page):
        super().__init__(page)
        self.locators = SharedLocators()
    
    def is_dashboard_displayed(self) -> bool:
        """Check if dashboard is displayed"""
        return self.is_element_visible(self.locators.dashboard_container)
    
    def get_welcome_message(self) -> str:
        """Get welcome message"""
        return self.page.locator(self.locators.welcome_message).text_content()
    
    def navigate_to_policies(self):
        """Navigate to policies section"""
        self.click_element(self.locators.policies_link)
    
    def navigate_to_claims(self):
        """Navigate to claims section"""
        self.click_element(self.locators.claims_link)
    
    def navigate_to_profile(self):
        """Navigate to profile section"""
        self.click_element(self.locators.profile_link)
    
    def logout(self):
        """Perform logout"""
        self.click_element(self.locators.logout_button)
    
    def get_active_policies_count(self) -> int:
        """Get number of active policies"""
        text = self.page.locator(self.locators.active_policies_count).text_content()
        return int(text) if text else 0
    
    def get_pending_claims_count(self) -> int:
        """Get number of pending claims"""
        text = self.page.locator(self.locators.pending_claims_count).text_content()
        return int(text) if text else 0

File: ts-framework/pages/dashboard_page.ts
import { Page } from '@playwright/test';
import { BasePage } from './base_page';
import { SharedLocators } from '../locators/shared_locators';

export class DashboardPage extends BasePage {
  private locators: SharedLocators;

  constructor(page: Page) {
    super(page);
    this.locators = new SharedLocators();
  }

  async isDashboardDisplayed(): Promise<boolean> {
    return await this.isElementVisible(this.locators.dashboard_container);
  }

  async getWelcomeMessage(): Promise<string> {
    return await this.page.locator(this.locators.welcome_message).textContent() || '';
  }

  async navigateToPolicies(): Promise<void> {
    await this.clickElement(this.locators.policies_link);
  }

  async navigateToClaims(): Promise<void> {
    await this.clickElement(this.locators.claims_link);
  }

  async navigateToProfile(): Promise<void> {
    await this.clickElement(this.locators.profile_link);
  }

  async logout(): Promise<void> {
    await this.clickElement(this.locators.logout_button);
  }

  async getActivePoliciesCount(): Promise<number> {
    const text = await this.page.locator(this.locators.active_policies_count).textContent();
    return text ? parseInt(text) : 0;
  }

  async getPendingClaimsCount(): Promise<number> {
    const text = await this.page.locator(this.locators.pending_claims_count).textContent();
    return text ? parseInt(text) : 0;
  }
}